scipy/special/_basic.py

```diff
@@ -3043,12 +3045,12 @@ def stirling2(N, K, exact=True):
                 num_iters = n - n_old
                 while num_iters > 0:
                     n_row.append(1)
+                    # traverse from back to remove second row
                     for j in range(len(n_row)-2, 1, -1):
                         n_row[j] = n_row[j]*j + n_row[j-1]
                     num_iters -= 1
                 snsk_vals[(n, k)] = n_row[k]
             else:
-                # value is in the same row as prev
```

Comment:
this comment was left over from when I had 2 arrays to do the calculation. Traversing from the back of the array removes the need for a second array, so I added a comment above to indicate what's happening here. O/w if I need to fix this in the future I'm unlikely to remember :)